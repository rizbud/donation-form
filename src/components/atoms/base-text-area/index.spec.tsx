import { render, screen } from "@testing-library/react";

import "@testing-library/jest-dom/extend-expect";

import userEvent from "@testing-library/user-event";

import { BaseTextArea } from ".";

describe("BaseTextArea", () => {
  it("should render a text area", () => {
    render(<BaseTextArea label="Label" />);

    expect(screen.getByRole("textbox")).toBeInTheDocument();
  });

  it("should render a hidden text area", () => {
    render(<BaseTextArea label="Label" hidden />);

    expect(screen.getByLabelText("Label")).toBeInTheDocument();
  });

  it("should render a text area with error message", () => {
    render(<BaseTextArea label="Label" error="Error" />);

    expect(screen.getByRole("textbox")).toBeInTheDocument();
    expect(screen.getByText("Error")).toBeInTheDocument();
  });

  it("should render a text area with placeholder", () => {
    render(<BaseTextArea label="Label" placeholder="Placeholder" />);

    expect(screen.getByPlaceholderText("Placeholder")).toBeInTheDocument();
  });

  it("should render a text area and write", async () => {
    render(<BaseTextArea label="Label" />);

    expect(screen.getByRole("textbox")).toBeInTheDocument();

    await userEvent.type(screen.getByRole("textbox"), "Text");

    expect(screen.getByRole("textbox")).toHaveValue("Text");
  });
});

// Snapshots
describe("BaseTextArea", () => {
  it("should render a text area", () => {
    const { container } = render(<BaseTextArea label="Label" />);

    expect(container).toMatchSnapshot();
  });

  it("should render a hidden text area", () => {
    const { container } = render(<BaseTextArea label="Label" hidden />);

    expect(container).toMatchSnapshot();
  });

  it("should render a text area with error message", () => {
    const { container } = render(<BaseTextArea label="Label" error="Error" />);

    expect(container).toMatchSnapshot();
  });

  it("should render a text area with placeholder", () => {
    const { container } = render(
      <BaseTextArea label="Label" placeholder="Placeholder" />,
    );

    expect(container).toMatchSnapshot();
  });
});
