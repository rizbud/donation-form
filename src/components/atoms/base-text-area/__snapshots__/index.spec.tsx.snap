// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseTextArea should render a hidden text area 1`] = `
<div>
  <div
    class="hidden mb-4"
  >
    <label
      class="block text-base font-semibold text-slate-800 mb-2"
    >
      Label
    </label>
    <div
      class="relative"
    >
      <textarea
        aria-invalid="false"
        aria-label="Label"
        class="placeholder:text-gray-400 text-black border-gray-300 block w-full rounded border p-4 text-base leading-6 enabled:focus:outline-none enabled:focus:ring-1 enabled:focus:border-primary enabled:focus:ring-primary"
        hidden=""
      />
    </div>
  </div>
</div>
`;

exports[`BaseTextArea should render a text area 1`] = `
<div>
  <div
    class="mb-4"
  >
    <label
      class="block text-base font-semibold text-slate-800 mb-2"
    >
      Label
    </label>
    <div
      class="relative"
    >
      <textarea
        aria-invalid="false"
        aria-label="Label"
        class="placeholder:text-gray-400 text-black border-gray-300 block w-full rounded border p-4 text-base leading-6 enabled:focus:outline-none enabled:focus:ring-1 enabled:focus:border-primary enabled:focus:ring-primary"
      />
    </div>
  </div>
</div>
`;

exports[`BaseTextArea should render a text area with error message 1`] = `
<div>
  <div
    class="mb-4"
  >
    <label
      class="block text-base font-semibold text-slate-800 mb-2"
    >
      Label
    </label>
    <div
      class="relative"
    >
      <textarea
        aria-invalid="true"
        aria-label="Label"
        class="placeholder:text-gray-400 text-black border-gray-300 block w-full rounded border p-4 text-base leading-6 enabled:focus:outline-none enabled:focus:ring-1 enabled:border-red-500 enabled:focus:ring-red-500"
      />
    </div>
    <span
      class="text-sm mt-1 text-red-500"
    >
      Error
    </span>
  </div>
</div>
`;

exports[`BaseTextArea should render a text area with placeholder 1`] = `
<div>
  <div
    class="mb-4"
  >
    <label
      class="block text-base font-semibold text-slate-800 mb-2"
    >
      Label
    </label>
    <div
      class="relative"
    >
      <textarea
        aria-invalid="false"
        aria-label="Label"
        class="placeholder:text-gray-400 text-black border-gray-300 block w-full rounded border p-4 text-base leading-6 enabled:focus:outline-none enabled:focus:ring-1 enabled:focus:border-primary enabled:focus:ring-primary"
        placeholder="Placeholder"
      />
    </div>
  </div>
</div>
`;
